src/NuGet.Clients/NuGet.VisualStudio.Common/ProjectType.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace NuGet.VisualStudio
+{
+    public static class ProjectType
+    {
+        private static readonly HashSet<string> Supported = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            VsProjectTypes.WebSiteProjectTypeGuid,
+            VsProjectTypes.CsharpProjectTypeGuid,
+            VsProjectTypes.VbProjectTypeGuid,
+            VsProjectTypes.CppProjectTypeGuid,
+            VsProjectTypes.JsProjectTypeGuid,
+            VsProjectTypes.FsharpProjectTypeGuid,
+            VsProjectTypes.NemerleProjectTypeGuid,
+            VsProjectTypes.WixProjectTypeGuid,
+            VsProjectTypes.SynergexProjectTypeGuid,
+            VsProjectTypes.NomadForVisualStudioProjectTypeGuid,
+            VsProjectTypes.TDSProjectTypeGuid,
+            VsProjectTypes.DxJsProjectTypeGuid,
+            VsProjectTypes.DeploymentProjectTypeGuid,
+            VsProjectTypes.CosmosProjectTypeGuid,
+            VsProjectTypes.ManagementPackProjectTypeGuid,
+        };
+
+        private static readonly HashSet<string> Unsupported = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            VsProjectTypes.LightSwitchProjectTypeGuid,
+            VsProjectTypes.InstallShieldLimitedEditionTypeGuid
+        };
+
+        // List of project types that cannot have binding redirects added
+        private static readonly string[] UnsupportedProjectTypesForBindingRedirects =
```

Comment:
```suggestion
        private static readonly HashSet<string> UnsupportedProjectTypesForBindingRedirects =
```

It's only used with `.Contains(...)`.